SIPS/sip-27.md

```diff
@@ -0,0 +1,64 @@
+---
+sip: 27
+title: Accounts Metadata Request
+status: Draft
+author: Olaf Tomalka (@ritave)
+created: 2024-09-17
+---
+
+## Abstract
+
+This SIP allows snaps to retrieve metadata related to accounts that exist in the extension.
+
+## Motivation
+
+The intention of this SIP is to allow snaps providing new accounts for new chains to be able to list all accounts when selecting one during transfers.
+
+## Specification
+
+> Indented sections like this are considered non-normative.
+
+> Usage of `CAIP-N` specifications, where `N` is a number, are references to [Chain Agnostic Improvement Proposals](https://github.com/ChainAgnostic/CAIPs).
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Snap Manifest
+
+This SIP specifies a permission named `keyring_listAccountsAll`. This permission grants a snap the ability to retrieve account metadata through an RPC call.
```

Comment:
What do you think of this?
```suggestion
This SIP specifies a permission named `keyring_listAllAccounts`. This permission grants a snap the ability to retrieve account metadata through an RPC call.
```

Reply:
I specifically don't want it to be mistaken with existing `keyring_listAccounts`, which only lists accounts managed by the Snap. IMHO, by appending it at the end it's more visible,

Reply:
I wonder if it makes sense to prefix this with `keyring_` in the first place. So far these APIs are only used by account Snaps.

Reply:
If we want to display _all_ accounts, we have to support how keyring manages and stores them in the extension, so this API mimics keyring internals almost 1:1.